pgmpy/inference/ExactInference.py

```diff
@@ -228,6 +228,47 @@ def induced_graph(self, elimination_order):
         >>> inference.induced_graph(['C', 'D', 'A', 'B', 'E'])
         <networkx.classes.graph.Graph at 0x7f34ac8c5160>
         """
+        eliminated_variables = set()
+        working_factors = {node: {factor for factor in self.factors[node]}
+                           for node in self.factors}
+
+        # The set of cliques should be in the induced graph
+        cliques = set()
+        for factors in working_factors.values():
+            for factor in factors:
+                cliques.add(tuple(factor.scope()))
+
+        # If the elimination_order list is incomplete, raise an error
+        if len(elimination_order) < len(self.variables):
```

Comment:
@pratyakshs: Please put all such check at the top in the method. You should also check if all the variables listed in `elimination_order` are in the model as in such cases we will get an error somewhere deep in code and will be very tough to trace back. So, it's better to have a check for that and raise a informative error.